storage/redis/client.go

```diff
@@ -235,6 +235,7 @@ func (client *Client) CompareAndSwap(ctx context.Context, key storage.Key, oldVa
 			_, err = tx.Pipelined(func(pipe redis.Pipeliner) error {
 				return put(ctx, pipe, key, newValue, client.TTL)
 			})
+			return Error.Wrap(err)
```

Comment:
We should not wrap the error here, as wrapping makes checking for `redis.TxFailedErr` unreliable.

Reply:
I pushed the change. This should fix the tests.